src/lib/djerba/render/database.py

```diff
@@ -0,0 +1,75 @@
+"""Interface with a CouchDB instance for JSON report documents"""
+
+import logging 
+import configparser 
+import requests
+import json
+import os
+
+import djerba.util.constants as constants
+import djerba.util.ini_fields as ini
+from datetime import datetime
+from djerba.util.logger import logger
+
+class Database(logger):
+    """Class to communicate with CouchDB via the API, eg. using HTTP GET/POST statements"""
+
+    def __init__(self, log_level=logging.WARNING, log_path=None):
+        self.logger = self.get_logger(log_level, __name__, log_path)
+        self.logger.info("Initializing Djerba database object")
+    
+    def Merge(self,dict1,dict2):
+        combined = {**dict1, **dict2}
+        return combined
+
+    """ Upload json to couchdb"""
+    def UploadFile(self,json_path):
+        time = datetime.now()
+        dt_couchDB = time.strftime("%d/%m/%Y %H:%M") 
+
+        with open(json_path) as report:
+            data = json.load(report)
+            config = data.get(constants.SUPPLEMENTARY).get(constants.CONFIG)
+            archive_base = config[ini.SETTINGS][ini.ARCHIVE_BASE]
+            archive_name = config[ini.SETTINGS][ini.ARCHIVE_NAME]           
+            db = config[ini.SETTINGS][ini.ARCHIVE_NAME]
+            base = config[ini.SETTINGS][ini.ARCHIVE_BASE]
+            if base[-1] != '/': 
+                base +='/'
+                self.logger.debug('Adding forward slash for url concatenation')
+            url = base + db
+            report_id = data["report"]["patient_info"]["Report ID"]
+            couch_info = {
+                '_id': '{}'.format(report_id), #DF val auto gen
+                'last_updated': '{}'.format(dt_couchDB),
+            }
+            upload = self.Merge(couch_info, data)
+            headers = {'Content-Type': 'application/json'}
+            submit = requests.post(url= url, headers= headers, json= upload)
+        
+        status = submit.status_code
+        uploaded = False
+        while uploaded == False:
```

Comment:
There are a few issues with this loop:
- It will loop forever unless the HTTP return code is exactly 201 or 409
- It will repeatedly hammer the DB server with retry attempts as often as Python can send them, which is a bad use of network resources and may even be interpreted as a malicious attack

Suggestions:
- Have an `else` clause to gracefully handly other HTTP status codes
- Use `time.sleep()` to delay a few seconds between retry attempts
- Have a maximum number of retry attempts (eg. 5), after which the loop will exit
- If an unrecoverable error is encountered, raise a `warning` log message to alert the user (or maybe `error`, depending if we want Djerba to continue after a failed archive attempt -- we can discuss this)